ext/vscode/src/extension.ts

```diff
@@ -0,0 +1,62 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { execSync } from 'child_process';
+
+export function activate(context: vscode.ExtensionContext) {
+    const didChangeEmitter = new vscode.EventEmitter<void>();
+
+    context.subscriptions.push(
+        vscode.lm.registerMcpServerDefinitionProvider('azureMcpProvider', {
+            onDidChangeMcpServerDefinitions: didChangeEmitter.event,
+            provideMcpServerDefinitions: async () => {
+                let npmPkg = '';
```

Comment:
Will it be more efficient to do this setup during extension activation? After the extension is activated, VS Code may list MCP configuration multiple times and we don't need to redo the configuration.